src/secret-storage.ts

```diff
@@ -162,6 +162,89 @@ interface Decryptors {
     decrypt: (ciphertext: IEncryptedPayload) => Promise<string>;
 }
 
+/**
+ * Interface provided by SecretStorage implementations
+ *
+ * Normally this will just be an {@link SecretStorage}, but for backwards-compatibility some methods allow other
+ * implementations.
+ */
+export interface ISecretStorage {
```

Comment:
Again, this isn't a great name: we're not supposed to call our interfaces `IFoo`.

(Maybe this should just be `SecretStorage`, and the new class should be `SecretStorageImpl` ?)

Reply:
What about `ServerSideSecretStorage` as a name for the server side secret storage implementation?

Reply:
As discussed elsewhere: I've gone for `ServerSideSecretStorage` for the interface and `ServerSideSecretStorageImpl` for the implementation.